packages/constellation/src/components/Compositions/Button/Button.tsx

```diff
@@ -0,0 +1,99 @@
+import cx from 'classnames'
+import React, { forwardRef } from 'react'
+
+import { IconCheck, SpinningLoader } from '../../Base'
+import {
+  ButtonState,
+  ButtonVariants,
+  Component as ComponentType,
+  Props,
+  Sizes,
+} from './Button.types'
+
+/**
+ * Buttons allow users to take actions, and make choices, with a single tap.
+ *
+ * When a `ref` prop is provided, it will be forwarded to the root element. Any
+ * other properties supplied will be provided to the root element (ie, the `as`
+ * prop value). This includes all appropriate HTML attributes or aria tags. This
+ * component may have its underlying DOM customized via the 'as' prop.
+ */
+
+const variantStyles: Record<ButtonVariants, string> = {
+  minimal:
+    'bg-background text-primary dark:text-blue-300 border border-grey-100 dark:hover:border-dark-500 dark:hover:bg-dark-800 hover:bg-blue-000 hover:border-blue-400 dark:border-dark-500 active:bg-blue-000 active:border-primary focus:bg-blue-000 focus:border-primary disabled:text-muted disabled:bg-background disabled:border-grey-100',
```

Comment:
the `dark:` selector is actually the default tailwind light and dark mode selector. In this branch the selector is `theme-dark:` and `theme-light:` but once the theme provider PR is merged it will be `mode-dark:` and `mode-light:` you might also run into the focus an active states from the light mode version overriding the dark mode colours. Some of them may be the `theme-light:` selector so they cant override the dark mode ones.